### ISSUE TITLE
Embed the HAProxy template file in the agent binary

### ISSUE THREAD

**Issue by ema-pe** [unimib-datAI/dfaas#14]
Currently in the agent configuration `dfaasagent.env` there are two keys that are always fixed with the same value in every file I found:

https://github.com/unimib-datAI/dfaas/blob/3a9fc2bf012c8e2c491e71b406bc75a5323d30d1/dfaasagent.env#L20-L21

These two file paths are the same in every configuration file, which means that the agent expects the same file and they are mandatory. To simplify the configuration file and reduce the number of unnecessary keys, we could embed these files into the binary using the Go [embed](https://pkg.go.dev/embed) package.

**Comment by aletundo** [comment]
> Currently in the agent configuration `dfaasagent.env` there are two keys that are always fixed with the same value in every file I found:
> 
> https://github.com/unimib-datAI/dfaas/blob/3a9fc2bf012c8e2c491e71b406bc75a5323d30d1/dfaasagent.env#L20-L21
> 
> These two file paths are the same in every configuration file, which means that the agent expects the same file and they are mandatory. To simplify the configuration file and reduce the number of unnecessary keys, we could embed these files into the binary using the Go [embed](https://pkg.go.dev/embed) package.

If these templates do not need to be modified at deployment time for any reason, yes it makes sense to embed.

**Comment by ema-pe** [comment]
> > Currently in the agent configuration `dfaasagent.env` there are two keys that are always fixed with the same value in every file I found:
> > https://github.com/unimib-datAI/dfaas/blob/3a9fc2bf012c8e2c491e71b406bc75a5323d30d1/dfaasagent.env#L20-L21
> > 
> > These two file paths are the same in every configuration file, which means that the agent expects the same file and they are mandatory. To simplify the configuration file and reduce the number of unnecessary keys, we could embed these files into the binary using the Go [embed](https://pkg.go.dev/embed) package.
> 
> If these templates do not need to be modified at deployment time for any reason, yes it makes sense to embed.

Yes, that is the point. I'm thinking if there is a need to change the template after deployment. In any case, the default templates should be embedded to provide some defaults.
